File: Backend/API/Data/Models/Shift.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using GPS.API.Interfaces;

namespace GPS.API.Data.Models
{
    public class Shift:IMustHaveTenant
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey(nameof(Bus))]
        public required int BusId { get; set; }
        public Bus? Bus { get; set; }
        [ForeignKey(nameof(Driver))]
        public required int DriverId { get; set; }
        public Driver? Driver { get; set; }
        public required DateOnly ShiftDate { get; set; }
        public required TimeOnly ShiftStartingTime { get; set; }
        public required TimeOnly ShiftEndingTime { get; set; }
        public string? TenantId { get; set; }
    }
}

File: Frontend2/Client/src/app/_models/shift.ts
import { Bus } from "./bus";
import { Driver } from "./driver";

export interface Shift {
    id: number;
    busId: number;
    bus: Bus;
    driverId: number;
    driver: Driver;
    shiftDate: string;
    shiftStartingTime: string;
    shiftEndingTime: string;
}
